content/6_Advanced/SCC.mdx

```diff
@@ -24,13 +24,24 @@ strongly connected component. We can compute these components using either
 <Resources>
 	<Resource source="CPH" title="17 - Strong Connectivity" />
 	<Resource source="Wikipedia" title="Kosaraju's Algorithm" url="https://en.wikipedia.org/wiki/Kosaraju%27s_algorithm" />
+	<Resource source="TC" title="Kosaraju's Algorithm for SCCs"
+	url="https://www.topcoder.com/thrive/articles/kosarajus-algorithm-for-strongly-connected-components"/>
 </Resources>
 
-<IncompleteSection>
+## Explanation
+Kosaraju's algorithm is a highly efficient method for identifying the strongly connected components (SCCs) of directed graphs. The process unfolds in two main phases.
 
-in-house explanation?
+In the first phase, the algorithm performs a depth-first search (DFS) on the original graph to determine the exit times of vertices, akin to creating a topological sort. As each vertex finishes processing, it is pushed onto a stack. In the next phase, nodes are processed off the stack in the order of their decreasing finish times. This ordering will be used in the second phase.
 
-</IncompleteSection>
+In the second phase, the algorithm conducts another DFS, this time on the transposed graph, where all edges are reversed. The DFS processes nodes according to the order defined by the stack from the first phase. By reversing the edges and following this specific order, each DFS run in this phase can identify all vertices of one SCC before moving on to the next. This ensures that the SCCs are accurately and efficiently determined.
+
+We know that the transposed graph will have the same SCCs as the original because reversing an edge doesn't change the reachability between two nodes. Since each SCC is maximal and disjoint, reversing edges doesn't break any connections in the original graph.
```

Comment:
> reversing an edge doesn't change the reachability between two nodes

misleading

Reply:
It's Ryan's suggestion from the previous commit, it's true in fact.

Reply:
a -> b
a <- b
say that again

Reply:
I am sure we didn't mean that. It's actually the "reachability between the two SCC components" does not change.
Sorry for misleading.